src/v2/context.in.rs

```diff
@@ -32,6 +32,17 @@ impl Context {
             Context::Dir(Path::new(&s_ref).to_owned())
         }
     }
+
+    /// Returns a new Context which is the same as the
+    /// this one, but without any subdirectory part
+    pub fn without_subdirectory(&self) -> Context {
```

Comment:
This is an _amazingly_ good choice of API. Thank you for figuring this out. Maybe it might be clearer if we named it `without_repository_subdirectory`?